tools/src/main/scala/scala/scalanative/codegen/Lower.scala

```diff
@@ -186,11 +199,8 @@ object Lower {
       case _                             => super.onVal(value)
     }
 
-    def genClassOf(buf: Buffer, node: ScopeInfo)(
-        implicit pos: Position): Val = {
-      val tpePtr = rtti(node).const
-      buf.call(toClassTy, toClassVal, Seq(Val.Null, tpePtr), Next.None)
-    }
+    def genClassOf(buf: Buffer, node: ScopeInfo)(implicit pos: Position): Val =
```

Comment:
This method no longer needs neither `Buffer` nor `Position` argument. There is a single usage of this method in `genVal`  that can now be inlined.
 
Also, usages of `genVal` can now be reverted to the previous `onVal` which does not need `Buffer` argument, however, it can be addressed is separate PR.